js/model/dinnerModel.js

```diff
@@ -1,52 +1,161 @@
+    },{
+      'name':'peaches',
+      'quantity':1,
+      'unit':'',
+      'price':4
+    }]
+  },{
+    'id':101,
+    'name':'MD 2',
+    'type':'main dish',
+    'image':'bakedbrie.jpg',
+    'description':"Here is how you make it... Lore ipsum...",
+    'ingredients':[{
+      'name':'ingredient 1',
+      'quantity':1,
+      'unit':'pieces',
+      'price':8
+    },{
+      'name':'ingredient 2',
+      'quantity':15,
+      'unit':'g',
+      'price':7
+    },{
+      'name':'ingredient 3',
+      'quantity':10,
+      'unit':'ml',
+      'price':4
+    }]
+  },{
+    'id':200,
+    'name':'Chocolat Ice cream',
+    'type':'dessert',
+    'image':'icecream.jpg',
+    'description':"Here is how you make it... Lore ipsum...",
+    'ingredients':[{
+      'name':'ice cream',
+      'quantity':100,
+      'unit':'ml',
+      'price':6
+    }]
+  }]
+
+  // Returns the dish that is(/are) on the (selected) menu for type
   this.getSelectedDish = function(type) {
-    //TODO Lab 1
+    var dishType;
+    this.selectedDishes.forEach(function(dish) {
+        if (dish["type"] === type) {
+           dishType = dish;
+        }
+    });
+    return dishType;
   }
 
-  //Returns all the dishes on the menu.
+  // Returns all the dishes on the (selected) menu.
   this.getFullMenu = function() {
-    //TODO Lab 1
+    return this.selectedDishes;
   }
 
-  //Returns all ingredients for all the dishes on the menu.
+  // Returns all ingredients for all the dishes on the (selected) menu.
   this.getAllIngredients = function() {
-    //TODO Lab 1
+    var ingredients = [];
+
+    this.selectedDishes.forEach(function(dish) {
+      dish["ingredients"].forEach(function(ingredient) {
+        ingredients.push(ingredient);
+      });
+    });
+
+    return ingredients;
   }
 
-  //Returns the total price of the menu (all the ingredients multiplied by number of guests).
+  // Returns the total price of the (selected) menu (all the ingredients
+  // multiplied by number of guests).
   this.getTotalMenuPrice = function() {
-    //TODO Lab 1
+    var totalPrice = 0;
+
+    this.selectedDishes.forEach(function(dish) {
+      dish["ingredients"].forEach(function(ingredient) {
+        totalPrice += ingredient["price"];
+      });
+    });
+
+    return totalPrice * this.getNumberOfGuests();
   }
 
-  //Adds the passed dish to the menu. If the dish of that type already exists on the menu
-  //it is removed from the menu and the new one added.
+  // Adds the passed dish to the (selected) menu. If the dish of that type
+  // already exists on the (selected) menu it is removed from the (selected)
+  // menu and the new one is added.
   this.addDishToMenu = function(id) {
-    //TODO Lab 1
+    var dishToAdd;
+    dishes.forEach(function(dish) {
+      if (dish["id"] === id) {
+        dishToAdd = dish;
+      }
+    });
+
+
+    var dishAlreadyInMenu = false;
+    for (var i=0; i<this.selectedDishes.length; i++) {
+      if (this.selectedDishes[i]["id"] === id) {
```

Comment:
if the dish is of the same TYPE, then replace it, thus should change the condition to:
if(this.selectedDishes[i].type===this.getDish(id).type)